ebu_tt_live/node/delay.py

```diff
@@ -0,0 +1,71 @@
+
+from .base import Node
+from datetime import timedelta
+from ebu_tt_live.bindings._ebuttdt import LimitedClockTimingType, FullClockTimingType
+import time
+import random
+
+
+class FixedDelayNode(Node):
+
+    _reference_clock = None
+    _sequence = None
+
+    def __init__(self, node_id, carriage_impl, reference_clock):
+        super(FixedDelayNode, self).__init__(node_id, carriage_impl)
+        self._reference_clock = reference_clock
+
+        random_int = random.randint(0, 9999999999)
+        new_identifier = 'sequence_{0}'.format(str(random_int))
+        self._sequence = new_identifier
+
+    def process_document(self, document):
+
+        # change the sequence identifier
+        document.sequence_identifier = self._sequence
+
+        # TODO: add an ebuttm:trace element to the document metadata
+
+        fixed_delay_int = 2
+
+        # document is explicitly timed: modify the document
+        if document.binding.body.begin or document.binding.body.end:
+
+            update_children_timing(document.binding, document.time_base, fixed_delay_int)
+            self._carriage_impl.emit_document(document)
+
+        # document is implicitly timed: pause a while, re-emit later
+        else:
+
+            time.sleep(fixed_delay_int)
```

Comment:
I'm not sure how twisted etc handle this: what happens if other documents arrive (or want to arrive) during this sleep? I wonder if there's a way to schedule the emission of the document for a known time instead of sleeping here, and if that would be better.